src/api/db/connect.ts

```diff
@@ -1,34 +1,32 @@
 import mongoose from 'mongoose';
 import { InternalServerError } from '../errors.js';
+import { Config } from '../../config/config.js';
 
 // TODO: consider using multiple connections (one per model)
 // to reduce risk of slow trains
 // https://mongoosejs.com/docs/connections.html#multiple_connections
 // https://thecodebarbarian.com/slow-trains-in-mongodb-and-nodejs
 
-let cachedConnectionPromise = null;
+let client: typeof mongoose | null = null;
 
-async function connectToDatabase(config) {
-  if (!cachedConnectionPromise) {
+async function connectToDatabase(config: Config) {
+  if (!client) {
     // If no connection promise is cached, create a new one.
     // We cache the promise instead of the connection itself to prevent race
     // conditions where connect is called more than once.
     const uri = config['/DB/MONGO_DB_URL'];
-    cachedConnectionPromise = await mongoose.connect(uri, {
+    client = await mongoose.connect(uri, {
       bufferCommands: false,
       // TODO: double check auto indexing is off in prod environment
-      autoIndex: process.env.STAGE !== 'prod'
+      autoIndex: process.env.STAGE !== 'prod',
     });
   }
 
   try {
-    const client = await cachedConnectionPromise;
     return client;
   } catch (err) {
-    throw new InternalServerError(err instanceof Error ? err : String(err));
+    throw new InternalServerError(err instanceof Error ? err.message : String(err));
```

Comment:
@alukach I'm curious about the reason for this change?

Reply:
The constructor of our `InternalServerError` expects a `string` argument, not an `Error` instance:

https://github.com/tnc-ca-geo/animl-api/blob/2e973b481f8b5f97378d83940f4d2025dac04088/src/api/errors.js#L8-L9